### PR TITLE
fixed ESC button when closing modals in sectioned.php

### PR DESCRIPTION
**Pull request by a22ahmom**
just before when you would press ESC you would be transported form sectioned.php to courseed.php when closing these popups with ESC, get canvas link, settings (JS-TEST template 3) and settings (JavaScript-Code) In the issue description it wasn't mentioned which specific popups this happened to but these three popups were however found and fixed.

Screenshot 2024-05-17 103934

### CODE REVIEW

**Review by b22chran — CHANGES_REQUESTED** [review]
The escape-function works fine when in the popup-windows that are described. However, when reading the issue it seems like it also should be closing the modal that appears when deleting items in sectioned.php instead of returning to coursed.php.
When this modal appears and pressing ESC it still returns to the courseed.php. I believe this was supposed to be fixed.
Skärmavbild 2024-05-17 kl  11 57 34

File: DuggaSys/sectioned.js [hunk 1]
```diff
@@ -617,7 +617,9 @@ function closeOpenPopupForm(){
     "#loginBox",
     "#loadDuggaBox",
     "#sectionHideConfirmBox",
-    "#sectionShowConfirmBox"
+    "#sectionShowConfirmBox",
+    "#canvasLinkBox",
+    "#editSection"
   ];
```
Comment (d22mikwe) [inline DuggaSys/sectioned.js:623]: Add "#toastContainer" here

File: Shared/toast.php [hunk 2]
```diff
@@ -124,6 +124,14 @@ function toast(type, text, duration, arguments = null) {
                 }
             }
         });
+
+        // Close the toast by pressing the ESC button
+        document.addEventListener('keydown', function (event) {
```
Comment (d22mikwe) [inline Shared/toast.php:129]: change to keyup

**Review by d22mikwe — CHANGES_REQUESTED** [review]
See comment. This should fix the issue

**Review by d22mikwe — APPROVED** [review]
Works now! LGTM!

**Pull request merged**